src/app/(sidebar)/smart-contracts/contract-explorer/components/ContractSpec.tsx

```diff
@@ -37,6 +47,47 @@ export const ContractSpec = ({
     isActive: Boolean(isActive && rpcUrl && wasmHash),
   });
 
+  const {
+    data: wasmBinary,
+    error: wasmBinaryError,
+    isLoading: isWasmBinaryLoading,
+    isFetching: isWasmBinaryFetching,
+    refetch: fetchWasmBinary,
+  } = useSEContractWasmBinary({ wasmHash, networkId });
+
+  const resetWasmBlob = useCallback(() => {
+    queryClient.resetQueries({
+      queryKey: ["useSEContractWasmBinary", networkId, wasmHash],
+    });
+  }, [networkId, queryClient, wasmHash]);
+
+  useEffect(() => {
+    if (wasmBinary) {
+      // Create blob
+      const blob = new Blob([wasmBinary], {
+        type: "application/octet-stream",
+      });
+
+      // Create a link element and trigger a download
+      const link = document.createElement("a");
+      link.href = URL.createObjectURL(blob);
+      link.setAttribute("download", `${wasmHash}.wasm`);
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+    }
+  }, [wasmBinary, wasmHash]);
+
+  useEffect(() => {
+    if (wasmBinaryError) {
+      // Automatically clear error message after 5 sec
+      delayedAction({
+        action: resetWasmBlob,
+        delay: 5000,
+      });
+    }
+  }, [resetWasmBlob, wasmBinaryError]);
```

Comment:
I couldn't find a way to clear this on specific events (unmount, inActive, etc.), so I opted to clear the message after 5 seconds. The UX seemed okay to me.